shoop/admin/modules/orders/json_order_creator.py

```diff
@@ -277,6 +276,22 @@ def create_source_from_state(self, state, creator=None, ip_address=None,
         if not self.is_valid:
             return None
 
+        if order_to_update:
+            # Preserve values that is not in state from original order
+            source.update(
+                creator=order_to_update.creator,
+                ip_address=order_to_update.ip_address,
+                orderer=order_to_update.orderer,
+                order_date=order_to_update.order_date,
+                status=order_to_update.status,
+                marketing_permission=order_to_update.marketing_permission,
+                customer_comment=order_to_update.customer_comment,
+                payment_data=(order_to_update.payment_data or {}),
+                shipping_data=(order_to_update.shipping_data or {}),
+                extra_data=(order_to_update.extra_data or {}),
+            )
```

Comment:
This should use OrderSource.update_from_order, since otherwise any new fields added to ordersource won't be updated by this code.